### ISSUE TITLE
controlling page number

### ISSUE THREAD

**Issue by enricoar** [xournalpp/xournalpp#1047]
Hallo,
again thanks for this amazing tool!

It would be useful to be able to "read off" the current page view  from a script or command line someway. For example  xounalpp could write the number of the current page view on a file.
This would be useful in connection with an additional pdf viewer (I have  xpdf in mind) that can be used in parallel. My idea is to have a script, bound to a key combination, that makes xpdf go to the same page as xournalpp.
The other way round (send xournalpp to the same page as xpdf) is easier as sending xournalpp to a desired page no can be done with a series of key combinations. Alternatively it would be of course easier if one could send xournalpp to a desired page from the command line. But this is probably more difficult to implement,

**Comment by LittleHuba** [comment]
Another issue we could use the signal channel mentioned in #1028 for.

**Comment by andreasb242** [comment]
@enricoar please explain why you would do something like that.

**Comment by enricoar** [comment]
@andreasb242: I often use xournalpp annotating a pdf and at the same time open a separate window in which I inspect the pdf alone. I use xpdf for this. There are several reasons for that: 1) I need to write on a certain page while looking at another one OR 2) With the pdf file I can follow hyperlinks  
From time to time I would like to move the pdf to the same page as xournalpp or vice-versa. For this I need to read somewhere the current xournalpp page

**Comment by andreasb242** [comment]
Ok, no Problem. Following link was also request in another Ticket (I think you requested it?)

Currently there is no API, but I would extend the scripting API, which would allow such things, but I need some free time ;-)

**Comment by rolandlo** [comment]
The Lua plugin API now has a way to get the current page number (for example) and also scrolling functions. Checkout the section on plugins on the [work-in-progress website](https://xournalpp.github.io/guide/plugins/). I have tested auto-scrolling in PR #2406. So combining these ideas it should be straightforward to come up with a solution for the problem at hand.
